### DISCUSSION TITLE
Plugin API: network.chat event server only?

### DISCUSSION THREAD

**Discussion by hawkerm** [OpenRCT2/OpenRCT2#19156] [Q&A] [answered]
Is the `network.chat` event only listened to on a server (`network.mode == "server"`) and not a client?

That's the behavior I'm currently seeing, but I was hoping to detect a chat message on a client as well for plugin communication across the network.

Currently registering like so:

```js
context.subscribe("network.chat", function(e) {
```

**Comment by hawkerm** [comment, marked as answer]
ah, just looking at the code, it seems this hook is only done on the server side when sending a message:

https://github.com/OpenRCT2/OpenRCT2/blob/f2b4d9dcf6477d575fe52ad9f0a5372be0272bae/src/openrct2/network/NetworkBase.cpp#L3783-L3793

Looks like on the client side where Chat is handled it should also have the hook to the plugin event here, eh?

https://github.com/OpenRCT2/OpenRCT2/blob/f2b4d9dcf6477d575fe52ad9f0a5372be0272bae/src/openrct2/network/NetworkBase.cpp#L2768-L2775

So this would be a bug/feature request?

**Reply by IntelOrca** [reply]
You can raise it as a bug if you like.

**Reply by hawkerm** [reply]
Awesome thanks, raised here with more details: https://github.com/OpenRCT2/OpenRCT2/issues/19158
